include/mbedtls/ecdsa.h

```diff
@@ -204,6 +235,51 @@ int mbedtls_ecdsa_write_signature_det( mbedtls_ecdsa_context *ctx,
 #endif /* MBEDTLS_DEPRECATED_REMOVED */
 #endif /* MBEDTLS_ECDSA_DETERMINISTIC */
 
+/**
+ * \brief           Convert a signature from ASN.1 to a raw concatenation
+ *                  of {r,s}
+ *
+ * \param sig       Signature to be converted
+ * \param ssize     Size of the signature
+ * \param byte_len  Length of a single number of the signature
+ * \param buf       Buffer pointer for the converted signature
+ * \param bufsize   Size of the passed buffer
+ * \param buflen    Size of the signature written to the buffer
+ *
+ * \note            The size of the buffer \c bufsize should be at least
+ *                  2*byte_len bytes long, otherwise this function will
+ *                  return an error.
+ *
+ * \return          0 if successful, or a MBEDTLS_ERR_ECP_BAD_INPUT_DATA or
+ *                  MBEDTLS_ERR_ASN1_LENGTH_MISMATCH error code
+ *
+ */
+int mbedtls_ecdsa_signature_to_raw( const unsigned char *sig,
+                            size_t ssize, uint16_t byte_len,
+                            unsigned char *buf, size_t bufsize,
+                            size_t* buflen );
+/**
+ * \brief           Convert a signature from numbers to ASN.1
+ *
+ * \param r         First number of the signature
+ * \param s         Second number of the signature
+ * \param sig       Buffer that will hold the signature
+ * \param slen      Length of the signature written
+ * \param ssize     Size of the sig buffer
+ *
+ * \note            The size of the buffer \c ssize should be at least
+ *                  `MBEDTLS_ECDSA_MAX_SIG_LEN(grp->pbits)` bytes long if
+ *                  the signature was produced from curve \c grp,
+ *                  otherwise this function will return an error.
+ *
+ * \return          0 if successful,
+ *                  or a MBEDTLS_ERR_MPI_XXX or MBEDTLS_ERR_ASN1_XXX error code
+ *
+ */
+int mbedtls_ecdsa_signature_to_asn1( const mbedtls_mpi *r,
```

Comment:
I'd prefer this to be the inverse of `mbedtls_ecdsa_signature_to_raw`, converting from a raw concatenation to ASN.1 in place.

Reply:
I also think this function should take bytes rather than MPIs as input, but I would prefer taking two pointers as input (allows the use case where r and s are concatenated in single buffer, but also other use cases).

For the same reason, as Gilles suggested elsewhere, I think the function should allow the input and output buffers to overlap, but not mandate that they are the same.

Reply:
Done as a new function - please re-review.